cesium/internal/unary/db.go

```diff
@@ -47,39 +47,30 @@ type DB struct {
 	Controller *controller.Controller[controlledWriter]
 	_idx       index.Index
 	mu         *openEntityCount
+	wrapError  func(error) error
+	closed     bool
 }
 
+var ErrDBClosed = core.EntityClosed("unary.db")
+
 func (db *DB) Index() index.Index {
 	if !db.Channel.IsIndex {
-		panic(fmt.Sprintf("[control.unary] - database %v does not support indexing", db.Channel.Key))
+		// inconceivable state
+		panic(fmt.Sprintf("channel %v is not an index channel", db.Channel.Key))
```

Comment:
Would be useful to include the names of these channels in the panic messages as well